### PR TITLE
Add Shared Image Gallery as an option

### PR DESCRIPTION
**Pull request by gavinfish**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java [hunk 1]
```diff
@@ -368,6 +370,31 @@ public AzureVMDeploymentInfo createDeployment(
             copyVariableIfNotBlank(tmp, properties, "osType");
             putVariableIfNotBlank(tmp, "image", template.getImage());
 
+            // Gallery Image is a special case for custom image, reuse the logic of custom image by replacing the imageId here
+            if (referenceType == ImageReferenceType.GALLERY) {
+                GalleryImageVersion  galleryImageVersion;
+                String galleryImageVersionStr = template.getGalleryImageVersion();
+                if (StringUtils.isBlank(galleryImageVersionStr)) {
```
Comment (yungezz) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java]: galleryImageDefinition/galleryResourceGroup empty check same as version?

File: src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-galleryImageDefinition.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Images are defined within a gallery and carry information about the image and requirements for using it internally. This includes whether the image is Windows or Linux, release notes, and minimum and maximum memory requirements. It is a definition of a type of image.
```
Comment (yungezz) [inline src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-galleryImageDefinition.html]: is this from azure doc?

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java (hunk 1 above)
Comment (gavinfish) [reply src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java]: done

File: src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-galleryImageDefinition.html (hunk 2 above)
Comment (gavinfish) [reply src/main/resources/com/microsoft/azure/vmagent/AzureVMAgentTemplate/help-galleryImageDefinition.html]: Yes, https://docs.microsoft.com/en-us/azure/virtual-machines/windows/shared-image-galleries.

File: src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java [hunk 3]
```diff
@@ -480,6 +512,22 @@ private boolean checkImageParameter(AzureVMAgentTemplate template) {
         return true;
     }
 
+    private GalleryImageVersion getGalleryImageLatestVersion(String galleryResourceGroup, String galleryName,
+                                                String galleryImageDefinition) {
+        PagedList<GalleryImageVersion> galleryImageVersions = azureClient.galleryImageVersions().listByGalleryImage(galleryResourceGroup, galleryName, galleryImageDefinition);
+        if (galleryImageVersions.isEmpty()) {
+            return null;
+        }
+        Collections.sort(galleryImageVersions, new Comparator<GalleryImageVersion>() {
```
Comment (yuwzho) [inline src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java]: need we do the sort here? We just wanna find the max one.

**Pull request merged**